changeset:   ce75724fe96acead18a459ebdcb5c8e5c32a1cfd  D9387914  (@)
parent:      3805657:eea4d4622d4f
user:        Orvid King <orvid@fb.com>
date:        Fri, 17 Aug 2018 15:53:37 -0700

    [Folly] [MSVC] Workaround ICE in MSVC 2017.8

    Summary:
    MSVC is having issues with this, so explicitly refer to `AllocTraits` via `Super`.

    Reviewers: yfeldblum

    Differential Revision: https://phabricator.intern.facebook.com/D9387914

diff -r eea4d4622d4f -r ce75724fe96a fbcode/folly/container/detail/F14Policy.h
--- a/fbcode/folly/container/detail/F14Policy.h	Fri Aug 17 15:45:37 2018 -0700
+++ b/fbcode/folly/container/detail/F14Policy.h	Fri Aug 17 15:53:37 2018 -0700
@@ -466,10 +466,11 @@
   using Super::kIsMap;
 
  public:
-  using ConstIter = ValueContainerIterator<typename AllocTraits::const_pointer>;
+  using ConstIter =
+      ValueContainerIterator<typename Super::AllocTraits::const_pointer>;
   using Iter = std::conditional_t<
       kIsMap,
-      ValueContainerIterator<typename AllocTraits::pointer>,
+      ValueContainerIterator<typename Super::AllocTraits::pointer>,
       ConstIter>;
 
   //////// F14Table policy
@@ -717,10 +718,11 @@
   using Super::kIsMap;
 
  public:
-  using ConstIter = NodeContainerIterator<typename AllocTraits::const_pointer>;
+  using ConstIter =
+      NodeContainerIterator<typename Super::AllocTraits::const_pointer>;
   using Iter = std::conditional_t<
       kIsMap,
-      NodeContainerIterator<typename AllocTraits::pointer>,
+      NodeContainerIterator<typename Super::AllocTraits::pointer>,
       ConstIter>;
 
   //////// F14Table policy
@@ -964,16 +966,18 @@
   using InternalSizeType = Item;
 
   using ConstIter =
-      VectorContainerIterator<typename AllocTraits::const_pointer>;
+      VectorContainerIterator<typename Super::AllocTraits::const_pointer>;
   using Iter = std::conditional_t<
       kIsMap,
-      VectorContainerIterator<typename AllocTraits::pointer>,
+      VectorContainerIterator<typename Super::AllocTraits::pointer>,
       ConstIter>;
-  using ConstReverseIter = typename AllocTraits::const_pointer;
-  using ReverseIter = std::
-      conditional_t<kIsMap, typename AllocTraits::pointer, ConstReverseIter>;
+  using ConstReverseIter = typename Super::AllocTraits::const_pointer;
+  using ReverseIter = std::conditional_t<
+      kIsMap,
+      typename Super::AllocTraits::pointer,
+      ConstReverseIter>;
 
-  using ValuePtr = typename AllocTraits::pointer;
+  using ValuePtr = typename Super::AllocTraits::pointer;
 
   //////// F14Table policy
 

